src/rqt_graph/dotcode.py

```diff
@@ -473,19 +615,33 @@ def generate_dotgraph(self,
 
         # for accumulating actions topics
         action_nodes = {}
+        # for accumulating image topics
+        image_nodes = {}
+        # for accumulating tf node connections
+        tf_connections = None
 
-        if graph_mode != NODE_NODE_GRAPH and (hide_single_connection_topics or hide_dead_end_topics or accumulate_actions):
+        if graph_mode != NODE_NODE_GRAPH and (hide_single_connection_topics or hide_dead_end_topics or accumulate_actions or group_tf_nodes or hide_tf_nodes or group_image_nodes or hide_dynamic_reconfigure):
             # maps outgoing and incoming edges to nodes
             node_connections = self._get_node_edge_map(edges)
 
             nt_nodes, edges = self._filter_leaf_topics(nt_nodes,
-                                         edges,
-                                         node_connections,
-                                         hide_single_connection_topics,
-                                         hide_dead_end_topics)
+                                        edges,
+                                        node_connections,
+                                        hide_single_connection_topics,
+                                        hide_dead_end_topics)
```

Comment:
Avoid arbitrary indentation  (even though it was that way before your patch :wink:). Either indent up to the open parenthesis or wrap directly after the parenthesis and indent all argument just one level / 4 spaces. The second option is preferred since it avoids the long indentation.

Same below.